tracing-jersey/src/main/java/com/palantir/tracing/jersey/TraceEnrichingFilter.java

```diff
@@ -124,12 +131,19 @@ private static Observability getObservabilityFromHeader(ContainerRequestContext
         }
     }
 
-    private enum ContainerResponseContextTagTranslator implements TagTranslator<ContainerResponseContext> {
+    private String getPathTemplate() {
+        return Optional.ofNullable(uriInfo)
+                .map(ExtendedUriInfo::getMatchedModelResource)
+                .map(Resource::getPath)
+                .orElse("(unknown)");
+    }
+
+    private enum FunctionalTagTranslator implements TagTranslator<Consumer<BiConsumer<String, String>>> {
         INSTANCE;
 
         @Override
-        public <T> void translate(TagAdapter<T> adapter, T target, ContainerResponseContext data) {
-            adapter.tag(target, TraceTags.HTTP_STATUS_CODE, Integer.toString(data.getStatus()));
+        public <T> void translate(TagAdapter<T> adapter, T target, Consumer<BiConsumer<String, String>> data) {
```

Comment:
Alright that took a second to grok. If you end up wanting a lambda a bit more, perhaps you want to ship that as an overload?

Reply:
Yep, this is a design I had considered, I wanted to try it out internally here to see how I like it -- Jersey already allocates a ton of data, so the lambda allocation is a drop in the bucket, I think most application code is similar so that such an api is reasonable.

If we add it as an overload, I'd end up writing new interfaces rather than the raw java functional interfaces, and allow chaining invocations to support expression lambdas e.g.
```java
sink -> sink.tag(TraceTags.HTTP_STATUS_CODE, Integer.toString(responseContext.getStatus()))
            .tag(TraceTags.HTTP_URL_PATH_TEMPLATE, getPathTemplate())
            .tag(TraceTags.HTTP_METHOD, requestContext.getMethod());
```

Reply:
That's cool